packages/lodestar/src/api/impl/validator/index.ts

```diff
@@ -176,6 +176,12 @@ export function getValidatorApi({chain, config, logger, metrics, network, sync}:
       notWhileSyncing();
       await waitForSlot(slot); // Must never request for a future slot > currentSlot
 
+      // Process the queued attestations in the forkchoice for correct head estimation
+      // forkChoice.updateTime() might have already been called by the onSlot clock
+      // handler, in which case this should just return.
+      chain.forkChoice.updateTime(slot);
```

Comment:
calling `forkChoice.updateTime()` multiple times would calling `processAttestationQueue()` multiple times.

I think we should do early return in `forkChoice.updateTime(slot)` if `this.fcStore.currentSlot >= slot`

Reply:
makes sense, updating! :+1: